src/nbl/asset/utils/CIESProfile.cpp

```diff
@@ -0,0 +1,145 @@
+#include "CIESProfile.h"
+
+using namespace nbl;
+using namespace asset;
+
+const CIESProfile::IES_STORAGE_FORMAT& CIESProfile::getIntegral() const
+{
+    size_t numHSamples = 2 * hAngles.size();
+    size_t numVSamples = 2 * vAngles.size();
+    double dTheta = core::PI<double>() / numVSamples;
+    double dPhi = 2 * core::PI<double>() / numHSamples;
+    double dThetaInAngle = MAX_VANGLE / numVSamples;
+    double dPhiInAngle = MAX_HANGLE / numHSamples;
+    double res = 0;
+    for (size_t i = 0; i < numVSamples; i++) {
+        for (size_t j = 0; j < numHSamples; j++) {
+            auto val = dPhi * dTheta * std::sin(dTheta * i) * sample(i * dThetaInAngle, j * dPhiInAngle);
+            res += val;
+        }
+    }
+    return res;
+}
+
+const CIESProfile::IES_STORAGE_FORMAT& CIESProfile::sample(double vAngle, double hAngle) const {
+    if (vAngle > vAngles.back()) return 0.0;
+    hAngle = hAngles.back() == 0.0
+        ? 0.0
+        : fmod(hAngle,
+            hAngles.back()); // when last horizontal angle is zero
+
+    // it's symmetric across all planes
```

Comment:
check for `hAngles.size()<2` and do an if statement, and have a SPECIAL way to handle it, set
```
hAngle = hAngles.front();
```

Reply:
`fmod` is not the correct operation, it radially repeats, the IES Viewer seems to mirror on a 90-270 degree test IES

Reply:
just stop with the FMOD, we'll patch 1995 profiles to 2002 spec outside

Reply:
resolved in following commit

Reply:
> 1995 header + [90,270] -> OTHER_HALF_SYMMETRIC, maps 89 -> 91 and 269 -> 271
I think you meant `271 -> 269`, right?

Reply:
> just stop with the FMOD, we need the cases turned into enums you handle with a switch

resolved anyway

> 1995 header + [90,270] -> OTHER_HALF_SYMMETRIC, maps 89 -> 91 and 269 -> 271
OTHER_HALF_SYMMETRIC horizontal angles patched to have HALF_SYMETRIC's range, 90 degrees offset is being saved for composition and order